consul/datadog_checks/consul/consul.py

```diff
@@ -483,6 +484,14 @@ def check_network_latency(self, agent_dc, main_tags):
         if len(nodes) == 1:
             self.log.debug("Only 1 node in cluster, skipping network latency metrics.")
         else:
+            socket_hostname = socket.gethostname()
+            if self.hostname != socket_hostname:
+                # Replace hostname references to use the agent determined hostname
+                # In EC2 environments, this will be `instance_id` instead, for example
+                for node in nodes:
+                    if node['Node'] == socket_hostname:
```

Comment:
in which case `node['Node'] != socket_hostname` ?

Reply:
In EC2 environments, when the agent chooses the AWS instance_id instead.

Reply:
In fact, hostname determination is an elaborate operation: https://github.com/DataDog/datadog-agent/blob/2ef29f5e2094279594594a6e754d74058f8f11d7/pkg/util/hostname.go#L132